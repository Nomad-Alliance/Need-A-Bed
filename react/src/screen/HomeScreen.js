import '../../config/i18n';

import React from 'react';
import { useTranslation } from 'react-i18next';
import { Image, ImageBackground, Pressable, Text, View } from 'react-native';

import Images from '../../config/Images';
import { globalSty } from '../style/globalStyle';

const HomeScreen = ({ navigation }) => {
  const { t } = useTranslation();

  return (
    <View style={globalSty.container}>
      <ImageBackground
        source={Images.NABBackgroundLogo}
        resizeMode="cover"
        style={globalSty.image}
      >
        <Image style={globalSty.logo} source={Images.logo} />
        <Text style={globalSty.texth1}>{t('HomeTran.welcome')}</Text>
        <Pressable
          title={t('HomeTran.homeBtn')}
          onPress={() => navigation.navigate('Location')}
          style={globalSty.Button}
        >
          <Text style={globalSty.btnText}>{t('HomeTran.homeBtn')}</Text>
        </Pressable>
        {/*<View style = {globalSty.space}/>*/}
        <Pressable
          title={t('HomeTran.feedback')}
          onPress={() => navigation.navigate('SurveyScreen')}
          style={globalSty.Button}
        >
          <Text style={globalSty.btnText}>{t('HomeTran.feedback')}</Text>
        </Pressable>
        <View style={globalSty.infoBtn}>
          <Pressable onPress={() => navigation.navigate('Info')}>
            <Image
              resizeMode="contain"
              style={globalSty.info}
              source={Images.info}
            />
          </Pressable>
        </View>
      </ImageBackground>
    </View>
  );
};

export default HomeScreen;
