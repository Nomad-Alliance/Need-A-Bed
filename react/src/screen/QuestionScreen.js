import '../../config/i18n';

import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import { useTranslation } from 'react-i18next';
import { ImageBackground, Pressable, Text, View } from 'react-native';

import CheckBox from '../../config/CheckBox';
import Images from '../../config/Images';
import { globalSty } from '../style/globalStyle';

const intialState = {
  LGBTQ: false,
  Veteran: false,
  Victim_DV: false,
  Children: false,
  Pregnant: false,
  Dog: false,
};

const QuestionScreen = ({ navigation }) => {
  const { t } = useTranslation();
  const [state, setState] = useState(intialState);
  //const [toggleButton,setToggleButton] = useState(false)
  const { handleSubmit } = useForm(); //here initialize handlesubmit from react-hooked
  const onsubmit = () => {
    navigation.navigate('Accessibility', { q1Data: state });
  };

  return (
    <View style={globalSty.container}>
      <ImageBackground source={Images.NABBackgroundLogo} resizeMode="cover" style={globalSty.image}>
        <View style={globalSty.pad}>
          <Text>{t('QuestionTran.text')}</Text>
          <View style={globalSty.CheckBoxForm}>
            <CheckBox
              onPress={(value) => setState({ ...state, LGBTQ: !state.LGBTQ })}
              title={t('QuestionTran.LGBTQ')}
              isChecked={state.LGBTQ}
            />
          </View>
          <View style={globalSty.CheckBoxForm}>
            <CheckBox
              onPress={(value) => setState({ ...state, Veteran: !state.Veteran })}
              title={t('QuestionTran.Veteran')}
              isChecked={state.Veteran}
            />
          </View>
          <View style={globalSty.CheckBoxForm}>
            <CheckBox
              onPress={(value) => setState({ ...state, Victim_DV: !state.Victim_DV })}
              title={t('QuestionTran.Victim_DV')}
              isChecked={state.Victim_DV}
            />
          </View>
          <View style={globalSty.CheckBoxForm}>
            <CheckBox
              onPress={(value) => setState({ ...state, Children: !state.Children })}
              title={t('QuestionTran.Childern')}
              isChecked={state.Children}
            />
          </View>
          <View style={globalSty.CheckBoxForm}>
            <CheckBox
              onPress={(value) => setState({ ...state, Pregnant: !state.Pregnant })}
              title={t('QuestionTran.Pregnant')}
              isChecked={state.Pregnant}
            />
          </View>
          <View style={globalSty.CheckBoxForm}>
            <CheckBox
              onPress={(value) => setState({ ...state, Dog: !state.Dog })}
              title={t('QuestionTran.Dog')}
              isChecked={state.Dog}
            />
          </View>
          <Pressable
            title={t('QuestionTran.contBtn')}
            onPress={handleSubmit(onsubmit)}
            style={globalSty.Button} //handlesubmit is from react hooked
          >
            <Text style={globalSty.btnText}>{t('QuestionTran.contBtn')}</Text>
          </Pressable>
        </View>
      </ImageBackground>
    </View>
  );
};

export default QuestionScreen;
