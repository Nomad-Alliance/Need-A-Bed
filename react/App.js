import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import React from 'react';

import ConfirmScreen from './src/screen/ConfirmScreen';
import DocScreen from './src/screen/Doc';
import Home from './src/screen/HomeScreen';
import InfoScreen from './src/screen/Info';
import Loc from './src/screen/locScreen';
import Loc_auto from './src/screen/locScreen_auto';
import Loc_man from './src/screen/locScreen_man';
import OSM_Screen from './src/screen/OSM_Screen';
import PlaceScreen from './src/screen/PlaceScreen';
import QuestionScreen from './src/screen/QuestionScreen';
import QuestionScreenTwo from './src/screen/QuestionScreen2';
import SurveyScreen from './src/screen/SurveyScreen';

const Stack = createStackNavigator();

export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName="Home">
        <Stack.Screen name="Home" component={Home} />
        <Stack.Screen name="Info" component={InfoScreen} />
        <Stack.Screen name="Doc" component={DocScreen} />
        <Stack.Screen name="Demographics" component={QuestionScreen} />
        <Stack.Screen name="Accessibility" component={QuestionScreenTwo} />
        <Stack.Screen name="Location" component={Loc} />
        <Stack.Screen name="LocAuto" component={Loc_auto} />
        <Stack.Screen name="LocMan" component={Loc_man} />
        <Stack.Screen name="SurveyScreen" component={SurveyScreen} />
        <Stack.Screen name="ShelterMap" component={OSM_Screen} />
        <Stack.Screen name="ShelterItem" component={PlaceScreen} />
        <Stack.Screen name="Confirm" component={ConfirmScreen} />
        <Stack.Screen name="PostSurvey" component={SurveyScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
