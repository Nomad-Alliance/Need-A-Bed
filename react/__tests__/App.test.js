import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from "../src/screen/HomeScreen";
import LocScreen from '../src/screen/locScreen';
import QuestionScreen from '../src/screen/QuestionScreen';
import QuestionScreenTwo from '../src/screen/QuestionScreen2';

import App from '../App';
import { Button } from 'react-native';
import CheckBox from '../config/CheckBox';

jest.useFakeTimers();

describe('<App />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(1);
  });
});
describe('<HomeScreen />', () => {
  it('HomeScreen button click generates next screen', () => {
    const fakeNavigation = {
      navigate: jest.fn(),
    };
    const inst = renderer.create(<HomeScreen navigation={fakeNavigation}/>);
    const button = inst.root.findByType(Button);

    button.props.onPress();

    expect(fakeNavigation.navigate).toBeCalledWith('Location')

  });
  it('HomeScreen renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('<LocScreen />', () => {
  it('Either Location Screen button click generates next screen', () => {

    const fakeNavigation = {
      navigate: jest.fn(),
    };
    const inst = renderer.create(<LocScreen navigation={fakeNavigation}/>);
    const button = inst.root.findAllByType(Button);

    button.at(0).props.onPress();

    expect(fakeNavigation.navigate).toBeCalledWith('LocAuto')

    button.at(1).props.onPress();

    expect(fakeNavigation.navigate).toBeCalledWith('LocMan')
  });
  it('LocScreen renders correctly', () => {
    const tree = renderer.create(<LocScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('<QuestionScreen />', () => {
  it('QuestionScreen button click generates next screen', () => {
    const fakeHandle = {
      handleSubmit: jest.fn(),
    };
    const fakeNavigation = {
      navigate: jest.fn(),
    };

    const inst = renderer.create(<QuestionScreen handleSubmit={fakeHandle} navigation={fakeNavigation}/>);
    const button = inst.root.findByType(Button);

    button.props.onPress();

    expect(fakeHandle)
  });
  it('QuestionScreen renders correctly', () => {
    const tree = renderer.create(<QuestionScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('<QuestionScreenTwo />', () => {
  it('QuestionScreen2 button click generates next screen', () => {
    const route = {
      params: {
        q1Data: 'my-test-id'
      }
    };
    const fakeHandle = {
      handleSubmit: jest.fn(),
    };
    const fakeNavigation = {
      navigate: jest.fn(),
    };
    //const form1Data = null;

    const inst = renderer.create(<QuestionScreenTwo route={route} handleSubmit={fakeHandle} navigation={fakeNavigation}/>);
    const button = inst.root.findByType(Button);

    button.props.onPress();

    expect(fakeHandle)
  });
  it('QuestionScreenTwo renders correctly', () => {
    const route = {
      params: {
        q1Data: 'my-test-id'
      }
    };
    const tree = renderer.create(<QuestionScreenTwo route={route}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
