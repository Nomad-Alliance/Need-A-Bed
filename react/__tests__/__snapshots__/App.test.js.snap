// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> HomeScreen renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <Image
      source={1}
      style={
        {
          "alignSelf": "center",
          "padding": 50,
          "width": 300,
        }
      }
    />
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 32,
          "fontWeight": "bold",
          "marginBottom": 16,
          "marginTop": 32,
          "textAlign": "center",
        }
      }
    >
      Find Local Shelters
    </Text>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          GET STARTED
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<LocScreen /> LocScreen renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <Image
      source={1}
      style={
        {
          "alignSelf": "center",
          "padding": 50,
          "width": 300,
        }
      }
    />
    <Text
      style={
        {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 40,
          "fontWeight": "bold",
          "lineHeight": 60,
          "textAlign": "center",
        }
      }
    >
      Enable Location Services?
    </Text>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Yes I Agree
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "black",
          "padding": 3,
        }
      }
    />
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          No, go Manual
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<QuestionScreen /> QuestionScreen renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <Text>
      Check All That Apply to You
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginHorizontal": 10,
          "paddingBottom": 20,
          "top": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginHorizontal": 5,
            "marginTop": 5,
            "width": 150,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text />
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
              "marginLeft": 5,
            }
          }
        >
          LGBTQ
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginHorizontal": 10,
          "paddingBottom": 20,
          "top": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginHorizontal": 5,
            "marginTop": 5,
            "width": 150,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text />
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
              "marginLeft": 5,
            }
          }
        >
          Veteran
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginHorizontal": 10,
          "paddingBottom": 20,
          "top": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginHorizontal": 5,
            "marginTop": 5,
            "width": 150,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text />
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
              "marginLeft": 5,
            }
          }
        >
          Victim of domestic violence
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginHorizontal": 10,
          "paddingBottom": 20,
          "top": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginHorizontal": 5,
            "marginTop": 5,
            "width": 150,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text />
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
              "marginLeft": 5,
            }
          }
        >
          Have custody of children who stay with me
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginHorizontal": 10,
          "paddingBottom": 20,
          "top": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginHorizontal": 5,
            "marginTop": 5,
            "width": 150,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text />
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
              "marginLeft": 5,
            }
          }
        >
          Pregnant
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginHorizontal": 10,
          "paddingBottom": 20,
          "top": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginHorizontal": 5,
            "marginTop": 5,
            "width": 150,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text />
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
              "marginLeft": 5,
            }
          }
        >
          Have a dog with me
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Continue
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<QuestionScreenTwo /> QuestionScreenTwo renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <Text>
      Check All That Apply to You
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginHorizontal": 10,
          "paddingBottom": 20,
          "top": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginHorizontal": 5,
            "marginTop": 5,
            "width": 150,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text />
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
              "marginLeft": 5,
            }
          }
        >
          Person of Color
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginHorizontal": 10,
          "paddingBottom": 20,
          "top": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginHorizontal": 5,
            "marginTop": 5,
            "width": 150,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text />
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
              "marginLeft": 5,
            }
          }
        >
          Disabled
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginHorizontal": 10,
          "paddingBottom": 20,
          "top": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginHorizontal": 5,
            "marginTop": 5,
            "width": 150,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text />
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
              "marginLeft": 5,
            }
          }
        >
          English is not my first language
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginHorizontal": 10,
          "paddingBottom": 20,
          "top": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginHorizontal": 5,
            "marginTop": 5,
            "width": 150,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text />
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
              "marginLeft": 5,
            }
          }
        >
          Deaf
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Continue
        </Text>
      </View>
    </View>
  </View>
</View>
`;
